keyboards/rastersoft/minitkl/minitkl.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 Raster Software Vigo
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "rgblight.h"
+
+void rgblight_call_driver(LED_TYPE *start_led, uint8_t num_leds) {
+
+    /* This is a SPI driver for the SK6812 and WS2812 RGB leds,
+     * for the ATmega32U.
+     * It sets SPI to 8MHz and sends 11100000 for a zero,
+     * or 11111110 for a one.
+     *
+     * The SPI interface is initialized in keyboard_post_init_user()
+     *
+     * It uses 44 bytes less than the WS2812 driver.
+     */
+
+    // Initialize SPI for the WS2812/SK6812 driver
+    // Set PB2 and PB0 as output (not doing it freezes the SPI)
+    DDRB |= (1<<PB2);
+    DDRB |= (1<<PB0);
+    /* Enable SPI, Master, set clock rate fck/2, MODE1 */
+    SPCR = (1<<SPE)|(1<<MSTR)|(1<<CPHA);
+    SPSR = 1<<SPI2X;
+    asm volatile (
+                    "mov  r24, %0\n\t"
+                    "add  r24, r24\n\t" // r24 = num_leds * 2
+                    "add   %0, r24\n\t" // %0 = num_leds * 3
+    "loop_byte:      ld   r24, Z+\n\t" // r24 contains the byte to send to the led
+                    "ldi  r25, 8\n\t" // 8 bits to send
+    "loop_bits:      ldi  r26, 0xE0\n\t" // 11100000 -> zero
+                    "sbrc r24, 7\n\t"
+                    "ldi  r26, 0xF8\n\t" // 11111110 -> one
+                    "out 0x2E, r26\n\t" // 0x2E -> SPDR
+                    "add  r24, r24\n\t" // rotate data one bit to the left
+    "wait_ready:     in   r27, 0x2D\n\t" // 0x2D -> SPSR
+                    "andi r27, 0x80\n\t" // check SPIF
+                    "breq wait_ready\n\t"
+                    "subi r25, 1\n\t" // are there more bits to send in this byte?
+                    "brne loop_bits\n\t"
+                    "subi  %0, 1\n\t"
+                    "brne loop_byte\n\t": :"a"(num_leds), "z"(start_led)
+    );
+}
```

Comment:
```suggestion
```

So from the PR checklist.
> no “old-school” or other low-level GPIO/I2C/SPI functions may be used – must use QMK abstractions unless justifiable 

As much as this is a saving, it's a very small amount. Especially when comparing with LTO enabled and I don't think it justifies it's inclusion. 

There's plenty of space to be saved else where that doesn't make the job of maintaining QMK more difficult.

With this removal, the file can deleted as it's not required to compile.

Reply:
@daskygit My idea was to migrate it into an interrupt-driven code, and that would save also CPU time...

Reply:
I also side with dasky on this one, that all this currently does is add maintenance burdens for negligible reward.

To move this keyboard PR forward, you should remove this implementation.

There is scope to provide alternative ws2812 implementations, but this would have to go via `develop` as a separate PR.

Reply:
Also, I agree in not using "old-school" functions in the generic code of QMK, but this is only for my keyboard, no other keyboards would use this code.

Reply:
The keyboard doesn't require this code to provide functionality that's not already covered by a core driver (albeit not as optimised). If it's allowed in this instance, it's then open to replication on further keyboard PRs.

An avenue has been suggested above by zvecr
> There is scope to provide alternative ws2812 implementations, but this would have to go via develop as a separate PR.

At which point I'd recommend opening a discussion on Discord about the possibility of it being implemented in core as an alternative ws2812 driver option.

Reply:
Ok, done.